Modix.Services/Utilities/EnumerableExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+
+namespace Modix.Services.Utilities
+{
+    public static class EnumerableExtensions
+    {
+        public static IEnumerable<(T Value, int Index)> AsIndexable<T>(this IEnumerable<T> source)
+        {
+            var index = 0;
+
+            foreach (var item in source)
+            {
+                yield return (item, index);
+                index++;
+            }
+        }
+
+        public static IEnumerable<(TFirst First, TSecond Second)> ZipOrDefault<TFirst, TSecond>(this IEnumerable<TFirst> first, IEnumerable<TSecond> second)
+        {
+            using (var e1 = first.GetEnumerator())
+            using (var e2 = second.GetEnumerator())
+            {
+                while (true)
```

Comment:
`while(e1.MoveNext() && e2.MoveNext())`?

Reply:
Doesn't that defeat the purpose? I wanted a Zip that went up to the length of the longest enumerable, not the shortest.

Reply:
I meant to do `||`, forgot that part when inverting the logic :)

Reply:
I'm not 100% convinced that's sufficient. I think for correctness I need to know which enumerator didn't move, since I can't guarantee that e.Current is `default` when the enumerator doesn't move

Reply:
In that case, if you're adamant about allowing this method for structs as well, then I rest my case.
If you're willing to change the constraints of the method to where T : class then we could change to the appropriate yield return e1.Current, e2.Current

Reply:
"I can't guarantee that e.Current is `default` when the enumerator doesn't move"

Reply:
Right, nevermind.